From ff6ba9321cb37d262740c3c4a7e904b6c75668d9 Mon Sep 17 00:00:00 2001
From: Pan Jiafei <Jiafei.Pan@freescale.com>
Date: Tue, 9 Sep 2014 11:10:13 +0800
Subject: [PATCH 791/987] capwap: fix frag OP stuck issue

It maybe a hw bug.

Signed-off-by: Pan Jiafei <Jiafei.Pan@freescale.com>
Change-Id: I7f328e7f325e886ee1e1c2bd468117b1b043dd9a
Reviewed-on: http://git.am.freescale.net:8181/21440
Reviewed-by: Jianhua Xie <jianhua.xie@freescale.com>
Tested-by: Shengzhou Liu <Shengzhou.Liu@freescale.com>
Reviewed-by: Shengzhou Liu <Shengzhou.Liu@freescale.com>
[Lu:Original patch taken from
QorIQ-SDK-V1.7-SOURCE-20141218-yocto.iso]
Signed-off-by: Jiang Lu <lu.jiang@windriver.com>
---
 drivers/net/ethernet/freescale/dpa/capwap/dpaa_capwap_domain.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/freescale/dpa/capwap/dpaa_capwap_domain.c b/drivers/net/ethernet/freescale/dpa/capwap/dpaa_capwap_domain.c
index 308c612..419bde0 100644
--- a/drivers/net/ethernet/freescale/dpa/capwap/dpaa_capwap_domain.c
+++ b/drivers/net/ethernet/freescale/dpa/capwap/dpaa_capwap_domain.c
@@ -1132,10 +1132,16 @@ int add_out_tunnel(struct dpaa_capwap_domain *capwap_domain,
 			e_FM_PCD_ACTION_EXACT_MATCH;
 		cc_node_param->extractCcParams.extractNonHdr.offset = 0;
 		cc_node_param->extractCcParams.extractNonHdr.size = 1;
-		cc_node_param->keysParams.numOfKeys = 0;
+		cc_node_param->keysParams.numOfKeys = 1;
 		cc_node_param->keysParams.keySize = 1;
 		cc_node_param->keysParams.maxNumOfKeys = 0;
 
+		cc_node_param->keysParams.keyParams[0].p_Key =
+			&cc_node_param->keysParams.keySize;
+		memcpy(&cc_node_param->keysParams.keyParams[0].
+				ccNextEngineParams,
+				fm_pcd_cc_next_engine_params,
+				sizeof(t_FmPcdCcNextEngineParams));
 		memcpy(&cc_node_param->keysParams.ccNextEngineParamsForMiss,
 				fm_pcd_cc_next_engine_params,
 				sizeof(t_FmPcdCcNextEngineParams));
-- 
1.9.1

